persistent/Database/Persist/Quasi.hs

```diff
@@ -538,11 +538,12 @@ lowerCaseSettings = defaultPersistSettings
 
 -- | Parses a quasi-quoted syntax into a list of entity definitions.
 parse :: PersistSettings -> Text -> [EntityDef]
-parse ps = parseLines ps . preparse
+parse ps = maybe [] (parseLines ps) . preparse
 
-preparse :: Text -> [Line]
+preparse :: Text -> Maybe (NonEmpty Line)
```

Comment:
`Maybe (NonEmpty a)` is equivalent to `[a]`. What's the purpose of this change?

Reply:
I think I'm just not sure about the benefit of *this* particular `[] -> NonEmpty` change. We don't appear to have any behavior that assumes or depends on `[]` being non-empty somewhere with this set of code.

But, it's also not a breaking change, and it seems like it makes sense semantically in the parsing section here.